packages/core/src/renderer/Renderer.ts

```diff
@@ -155,23 +201,83 @@ export const RenderInteractive = async (
   const onDrag = (id: string, dx: number, dy: number) => {
     updateState(dragUpdate(state, id, dx, dy));
   };
-  for (const shape of state.computeShapes(state.varyingValues)) {
-    svg.appendChild(
-      await DraggableShape(
-        {
-          shape,
-          labels: state.labelCache,
-          canvasSize: state.canvas.size,
-          pathResolver,
-        },
-        onDrag,
-        svg
-      )
-    );
-  }
+  const shapes = state.computeShapes(state.varyingValues);
+  const graph = makeGroupGraph(shapes);
+  await RenderGroupGraphInteractive(
+    graph,
+    svg,
+    {
+      labels: state.labelCache,
+      canvasSize: state.canvas.size,
+      pathResolver,
+    },
+    onDrag
+  );
   return svg;
 };
 
+const RenderGroup = async (
+  name: string,
+  graph: GroupGraph<Shape>,
+  shapeProps: {
+    labels: LabelCache;
+    canvasSize: [number, number];
+    pathResolver: PathResolver;
+  }
+): Promise<SVGGElement> => {
+  const elem = document.createElementNS("http://www.w3.org/2000/svg", "g");
+  const orderedChildren = graph.children(name).sort((a, b) => {
+    return graph.node(a).index - graph.node(b).index;
+  });
+  for (const childName of orderedChildren) {
+    const childSvg = await RenderGroupGraphNode(childName, graph, shapeProps);
+    elem.appendChild(childSvg);
+  }
+  const node = graph.node(name);
+  attrAutoFillSvg(node.shape, elem, [...attrTitle(node.shape, elem), "shapes"]);
+  return elem;
+};
+
+const RenderGroupGraphNode = async (
+  name: string,
+  graph: GroupGraph<Shape>,
+  shapeProps: {
+    labels: LabelCache;
+    canvasSize: [number, number];
+    pathResolver: PathResolver;
+  }
+): Promise<SVGElement> => {
+  const node = graph.node(name);
+  if (node.shape.shapeType === "Group") {
+    const outSvg = await RenderGroup(name, graph, shapeProps);
+    graph.delNode(name);
+    return outSvg;
+  } else {
+    const outSvg = await shapeMap[node.shape.shapeType]({
+      ...shapeProps,
+      shape: node.shape,
+    });
+    graph.delNode(name);
+    return outSvg;
+  }
+};
+
+const RenderGroupGraph = async (
```

Comment:
Done - I'm actually passing in an optional `interactiveProp` with everything needed for interactivity. If `interactiveProp` is undefined, we render statically. Otherwise we render interactively.